File: python/48.Rotate Image.py
class Solution:
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        n=len(matrix)

        for i in range(n//2):
        	j=n-1-i
        	for k in range(n):
        		tmp=matrix[i][k]
        		matrix[i][k]=matrix[j][k]
        		matrix[j][k]=tmp
       	
       	for i in range(n):
       		for j in range(i+1,n):
       			tmp=matrix[i][j]
       			matrix[i][j]=matrix[j][i]
       			matrix[j][i]=tmp

       	#return matrix


print(Solution().rotate([[1,2,3],[4,5,6],[7,8,9]]))

File: cpp/48.Rotate Image.cpp
#include <vector>
using namespace std;


class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int low = 0;
        int high = matrix.size()-1;
        int tmp;
        while(high - low >= 1){
            for (int i=low; i<high; i++){
                tmp = matrix[low][i];
                matrix[low][i] = matrix[high-i+low][low];
                matrix[high-i+low][low] = matrix[high][high-i+low];
                matrix[high][high-i+low] = matrix[i][high];
                matrix[i][high] = tmp;
            }
            --high;
            ++low;
        }
    }
};
